src/libs/PolicyUtils.ts

```diff
@@ -407,6 +411,39 @@ function getSubmitToAccountID(policy: OnyxEntry<Policy>, employeeAccountID: numb
     return getAccountIDsByLogins([employee.submitsTo ?? defaultApprover])[0];
 }
 
+function getSubmitToEmail(policy: OnyxEntry<Policy>, employeeAccountID: number): string {
+    const submitToAccountID = getSubmitToAccountID(policy, employeeAccountID);
+    return getLoginsByAccountIDs([submitToAccountID])[0] ?? '';
+}
+
+function getForwardsToAccount(policy: OnyxEntry<Policy>, employeeEmail: string, reportTotal: number): string {
+    if (!hasAdvancedApprovalEnabled(policy)) {
+        return '';
+    }
+
+    const employee = policy?.employeeList?.[employeeEmail];
+    if (!employee) {
+        return '';
+    }
+
+    const positiveReportTotal = Math.abs(reportTotal);
+    if (employee.approvalLimit && positiveReportTotal > employee.approvalLimit) {
+        return employee.overLimitForwardsTo ?? '';
+    }
+    return employee.forwardsTo ?? '';
+}
+
+function getApprovalChain(policy: OnyxEntry<Policy>, employeeAccountID: number, reportTotal: number): string[] {
+    const approvalChain: string[] = [];
+    let nextApproverEmail = getSubmitToEmail(policy, employeeAccountID);
+
+    while (!approvalChain.includes(nextApproverEmail)) {
```

Comment:
In our backend code we also have a check to only enter this loop if `nextApproverEmail` is truthy / is not just an empty string, don't we need that here? like this?

```suggestion
    while (nextApproverEmail && !approvalChain.includes(nextApproverEmail)) {
```

Reply:
Then we could probably get rid of the filter at the end?

Reply:
good suggestion 👍